### PR TITLE
Disable TLC config files by default

### LINKED ISSUE #1673: CLI option to ignore local cfg file
**Is your feature request related to a problem? Please describe.**
I am unable to tell Apalache to ignore a local TLC config file with the name ```<file>.cfg```. The cfg file is there because I with to support both TLC and Apalache.

**Describe the solution you'd like**
A command line flag or similar to tell Apalache not to look for a cfg file

**Describe alternatives you've considered**
Changing the name of the cfg file and using ```-config``` to change the location that TLC uses but this requires modification to the existing workflow for TLC

**Additional context**
Not an important feature but would be nice to have

### PR DESCRIPTION
**Pull request by konnov**
Closes #1673. This PR flips the behavior of Apalache w.r.t. TLC configuration files. The new version never loads a TLC config file, unless it is passed via `--config=...`.

Since this PR changes the default behavior, it is a breaking change, which requires a version bump.

- [ ] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Review by thpani — COMMENTED** [review]
The default value mentions below probably also need to be updated?

```
docs/src/apalache/running.md
27:    - `--config` specifies the [TLC configuration file](./tlc-config.md), the default name is `<myspec>.cfg`

mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/AbstractCheckerCmd.scala
13:        "default: <file>.cfg, or none if <file>.cfg not present")
```

Edit: Or should we default to `<myspec>.cfg` on `--config` (with an empty argument)?

File: tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala [hunk 1]
```diff
@@ -141,17 +143,28 @@ class ConfigurationPassImpl @Inject() (
...
+      // Hence, warn the user that their config is not loaded by default.
+      val stem = FilenameUtils.removeExtension(options.getOrError[String]("parser", "filename"))
+      val defaultConfig = new File(stem + ".cfg")
+      if (defaultConfig.exists()) {
+        val msg = s"  > Ignoring the accompanying TLC config file. To enable it, pass --config=$defaultConfig."
+        logger.info(msg)
       }
-    val basename = FilenameUtils.getName(filename)
+      // return immediately without going all the way down
+      return List()
```
Comment (thpani) [inline tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala:159]: I had to jump up to the Scaladoc to understand what `List()` is here, so probably this is better:

```suggestion
      // return immediately, with no additional declarations
      return List()
```

**Comment by konnov** [comment]
> The default value mentions below probably also need to be updated?
> 
> ```
> docs/src/apalache/running.md
> 27:    - `--config` specifies the [TLC configuration file](./tlc-config.md), the default name is `<myspec>.cfg`
> 
> mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/AbstractCheckerCmd.scala
> 13:        "default: <file>.cfg, or none if <file>.cfg not present")
> ```
> 
> Edit: Or should we default to `<myspec>.cfg` on `--config` (with an empty argument)?

Good point. I think we should not have any defaults :)

**Review by thpani — APPROVED** [review]
Up to the 2 mentions of default values above, LGTM!

File: tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala [hunk 2]
```diff
@@ -141,17 +143,28 @@ class ConfigurationPassImpl @Inject() (
...
-      } else {
-        configFilename
+    if (configFilename.isEmpty) {
+      // The older versions of apalache were loading a TLC config file of the same basename as the spec.
+      // We have flipped this behavior in version 0.25.0.
+      // Hence, warn the user that their config is not loaded by default.
+      val stem = FilenameUtils.removeExtension(options.getOrError[String]("parser", "filename"))
+      val defaultConfig = new File(stem + ".cfg")
+      if (defaultConfig.exists()) {
+        val msg = s"  > Ignoring the accompanying TLC config file. To enable it, pass --config=$defaultConfig."
```
Comment (Kukovec) [inline tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala:155]:
```suggestion
        val msg = s"  > TLC config file found in specification directory. To enable it, pass --config=$defaultConfig."
```

**Review by Kukovec — APPROVED** [review]
💯

**Pull request merged**